packages/server/src/fhir/search.ts

```diff
@@ -79,1146 +79,1157 @@ export interface ChainedSearchParameter {
   chain: ChainedSearchLink[];
 }
 
-export async function searchImpl<T extends Resource>(
-  repo: Repository,
-  searchRequest: SearchRequest<T>
-): Promise<Bundle<T>> {
-  validateSearchResourceTypes(repo, searchRequest);
-
-  // Ensure that "count" is set.
-  // Count is an optional field.  From this point on, it is safe to assume it is a number.
-  if (searchRequest.count === undefined) {
-    searchRequest.count = DEFAULT_SEARCH_COUNT;
-  } else if (searchRequest.count > maxSearchResults) {
-    searchRequest.count = maxSearchResults;
-  }
+export class RepositorySearch {
```

Comment:
This is easiest to review with "ignore whitespace", since most of these changes are just indentation